Common/Tests/Utilities/FileUtils.cs

```diff
@@ -167,6 +170,14 @@ public static class FileUtils {
             foreach (var newFile in newFiles) {
                 var copyFrom = Path.Combine(sourceDir, newFile);
                 var copyTo = Path.Combine(destDir, newFile);
+
+                if (tryHardLinkFirst) {
```

Comment:
Wouldn't it be better to hardlink the whole directory? If hard linking fails for one file, it will probably fail for the others, and conversely, if it succeeds for one, it probably succeeds for all, so I don't think there's any real benefit in trying to hardlink-or-copy on a per-file basis.